py_ecc/bls/point_compression.py

```diff
@@ -30,6 +31,52 @@
 )
 
 
+#
+# The most-significant three bits of a G1 or G2 encoding should be masked away before
+# the coordinate(s) are interpreted.
+# These bits are used to unambiguously represent the underlying element
+# The format: (c_flag, b_flag, a_flag, x)
+# https://github.com/zcash/librustzcash/blob/6e0364cd42a2b3d2b958a54771ef51a8db79dd29/pairing/src/bls12_381/README.md#bls12-381-instantiation  # noqa: E501
+#
+def get_c_flag(z: int) -> bool:
+    """
+    The most significant bit.
+    """
+    return bool((z % POW_2_384) // POW_2_383)
+
+
+def get_b_flag(z: int) -> bool:
+    """
+    The second-most significant bit.
+    """
+    return bool((z % POW_2_383) // POW_2_382)
+
+
+def get_a_flag(z: int) -> bool:
+    """
+    The third-most significant bit.
+    """
+    return bool((z % POW_2_382) // POW_2_381)
+
+
+def is_point_at_infinity(z1: int, z2: Optional[int] = None) -> bool:
+    return (z1 % POW_2_381 == 0) and (
+        z2 is None or (z2 is not None and z2 == 0)
+    )
+
+
+def validate_point_at_infinity(z1: int, a_flag: bool, z2: Optional[int] = None) -> None:
+    """
+    If z2 is None, the given z1 is a G1 point.
+    Else, (z1, z2) is a G2 point.
+    """
+    if a_flag == 0:
```

Comment:
since a_flag is bool
```suggestion
    if not a_flag:
```